### PR TITLE
6 moar defaults

### PR DESCRIPTION
**Pull request by jhogue**
So, some big changes to review. I suggest pull down the latest and look at each file. 

My intent was to add in some useful functions and mixins, PLUS some very very base styles needed to run the Wireframe package out of the box. These can easily be erased later, but the files (header, footer, content) are useful to keep in the package.

### CODE REVIEW

File: base/_typography.scss [hunk 1]
```diff
@@ -1,231 +1,225 @@
...
+	color: palette(mono, 75);
 	text-decoration: none;
-}
 
-a:hover,
-a:active {
-	color: palette(mono, black);
-	text-decoration: underline;
-}
+	@include touch-hover() {
```
Comment (bendoh) [inline base/_typography.scss:28]: i think a level of indentation on lines 30 and 31 is missing

**Comment by bendoh** [comment]
Looks fine to me, just have one comment about indentation in one file.

File: base/_functions.scss [hunk 2]
```diff
@@ -12,3 +12,46 @@
 	}
 	@return $map;
 }
+
+
+// Bourbon changes (assumed) px values to EM or REM.
+// Here we change a unitless value to px
+// @jhogue
+@function px($value) {
```
Comment (johncionci) [inline base/_functions.scss:20]: Is this different from http://bourbon.io/docs/#strip-units

File: base/_functions.scss [hunk 3]
```diff
@@ -12,3 +12,46 @@
...
+// @usage: For a corresponding text size (from base/_textsize.scss): rwd-size(element,group)
+@function bp($size: 'base') {
+	@return em( map-get( $breakpoints, $size ) );
+}
+
+@function bppx($size: 'base') {
+	@return px( map-get( $breakpoints, $size ) );
+}
+
+@function rwd-size($elem, $group: 'base') {
```
Comment (johncionci) [inline base/_functions.scss:42]: I think we can consolidate this now that we have the sass `deep-map-get()` helper, but that could come later.

File: base/_mixins.scss [hunk 4]
```diff
@@ -68,3 +69,21 @@
...
+	// scss-lint:disable DuplicateProperty
+	font-size: px($value);
+	font-size: rem($value);
+}
+
+
+// Bring back a border-radius mixin, if only so we can define a default
+// They have not deprecated their individual mixins, i.e. border-radius-top(), etc…
+// @jhogue
+@mixin border-radius($radius: $default-border-radius) {
```
Comment (johncionci) [inline base/_mixins.scss]: Built in to bourbon http://bourbon.io/docs/#border-radius

File: base/_typesizes.scss [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+);
+
+
+// ! ===== Mixins to support pulling out type sizes at breakpoints =====
+
+//
+// A deep function used by other mixins
+// Check each $key-s by name and treats some of them special
+// @jhogue
+//
```
Comment (johncionci) [inline base/_typesizes.scss:69]: Can we just add a @uses here for the rem-fontsize mixin , since its dependent.

File: base/_typography.scss [hunk 6]
```diff
@@ -1,231 +1,225 @@
...
 	font-family: $font-sans;
-	font-size: $paragraph;
+	font-weight: normal;
 	line-height: $line-height;
 }
 
-a:link,
-a:visited {
-	color: palette(mono, black);
+a {
```
Comment (johncionci) [inline base/_typography.scss:22]: I have a link mixin that we can use to add to this, once we merge this in.

File: base/_typography.scss [hunk 7]
```diff
@@ -1,231 +1,225 @@
...
+h4,
+.delta {
+	@include one-element-size(h4);
 }
 
-h6 {
-	font-size: $heading-6;
+h5,
+.echo {
+	@extend %base-rwd-size;
```
Comment (johncionci) [inline base/_typography.scss:90]: Are we still using @extends on a limited basis?

File: base/_typography.scss [hunk 8]
```diff
@@ -238,11 +232,172 @@ table {
...
+
+abbr[title] {
+	font-size: 75%;
+	font-weight: 700;
+	letter-spacing: 0.125em;
+	text-transform: uppercase;
+}
+
+b,
+strong { font-weight: 700; }
```
Comment (johncionci) [inline base/_typography.scss:279]: Is _bold_ the same as 700 when using a Google font? Google seems to always use the numerical weight definition.

File: base/_typography.scss [hunk 9]
```diff
@@ -238,11 +232,172 @@ table {
...
+	display: inline-block;
+	font-style: italic;
+	padding: 0 .125em;
+}
+
+// Proper code blocks need to be surrounded by a <pre>. They work together.
+pre {
+	// scss-lint:disable ImportantRule
+	white-space: pre-wrap;
+	white-space: -moz-pre-wrap !important;	// 1999+ Mozilla
```
Comment (johncionci) [inline base/_typography.scss]: Shouldn't need any prefix usage here, unless it has to do with the !important rule. I know that even with brunch we get Autoprefixer.

File: base/_typography.scss [hunk 10]
```diff
@@ -238,11 +232,172 @@ table {
...
+	padding: .125em .25em 0;
+}
+
+code,
+kbd,
+samp {
+	// scss-lint:disable DuplicateProperty
+	background: palette(mono, 10);
+	background: rgba(#000, .05);
+	border-radius: 4px;
```
Comment (johncionci) [inline base/_typography.scss]: border-radius var?

File: base/_typography.scss [hunk 11]
```diff
@@ -238,11 +232,172 @@ table {
...
 		}
+
+		// Nested
+		q::before { content: "‘"; }
+
+		q::after { content: "’"; }
 	}
 }
+
+.typography {
```
Comment (johncionci) [inline base/_typography.scss:368]: Are we using this class to define anything else or is this just for the styleguide?

File: base/_variables.scss [hunk 12]
```diff
@@ -1 +1,15 @@
-// @file variables.scss
+// Variables throughout the project
```
Comment (johncionci) [inline base/_variables.scss:1]: You think these belong globally? I think that we still need to look at further extrapolating out the responsive/typography stuff maybe into a base library as its a pretty significant piece of out scaffold.

File: component/_buttons.scss [hunk 13]
```diff
@@ -1 +1,57 @@
...
+	}
+	color: palette(mono, 66);
+	text-align: center;
+	
+	@include touch-hover() {
+		background-color: palette(mono, 10);
+		color: palette(mono, 75);
+	}
+
+	&--primary {
```
Comment (johncionci) [inline component/_buttons.scss:38]: I have a button mixin also to add into scaffold. I would try as much as possible to stay away from using predefined classes as for the most part we may not have access to code/classes without some programmatic implementations.

My mixin basically take a parameter such as `$primary` that could be a pplied to any buttons via css.

File: component/_content.scss [hunk 14]
```diff
@@ -0,0 +1,83 @@
+.content {
```
Comment (johncionci) [inline component/_content.scss:1]: Same here as for using default classes.

File: component/_footer.scss [hunk 15]
```diff
@@ -0,0 +1,13 @@
+.mainfoot {
```
Comment (johncionci) [inline component/_footer.scss]: Same here as for using default classes.

File: component/_forms.scss [hunk 16]
```diff
@@ -55,23 +88,49 @@ input[type="file"] {
...
+input[type="number"] { -webkit-appearance: textfield; }
+
+
+// Forms with a class .form and some helpful patterns that come up often
+.form {
+	@include clearfix();
 
-#{$all-buttons-focus},
-#{$all-buttons-active} {}
+	// Single line input and button pattern. Button needs to be in the markup first
```
Comment (johncionci) [inline component/_forms.scss]: Same here as for using default classes.

File: component/_header.scss [hunk 17]
```diff
@@ -0,0 +1,18 @@
+.mainhead {
```
Comment (johncionci) [inline component/_header.scss]: Same here as for using default classes.

File: component/_navigation.scss [hunk 18]
```diff
@@ -1 +1,28 @@
 // @file navigation.scss
+.mainnav {
```
Comment (johncionci) [inline component/_navigation.scss]: Same here as for using default classes.

**Comment by johncionci** [comment]
Lots of general comments concerning items that we already include with Bourbon, also adding default class declarations for nav, typography, etc.

**Comment by jhogue** [comment]
@johncionci Can we reach a happy medium where we have some default classes that can easily be changed so that at first checkout, I can have the wireframes (Grey Matter) working in the most basic way?

**Comment by johncionci** [comment]
Is GreyMatter straight HTML or will that be CMS dependent? Let me check what the element classes are at the most generic level that hopefully each CMS outputs.

File: component/_footer.scss [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+	background-color: palette(mono,75);
+	color: palette(mono,10);
+	padding: $vert-line-height 0;
+
+	a {
+		color: palette(mono,10);
+		
+		@include touch-hover() {
```
Comment (johncionci) [inline component/_footer.scss]: Should every `a` get `touch-hover()` ?

**Comment by johncionci** [comment]
... and of course none of the element classes or anything matchup between the two CMS's.
Maybe we just add the classes as they are output in the CMS so we would have:

```
.site-footer, // covers WordPress
.region-footer {} // covers Drupal
```

File: component/_footer.scss (hunk 19 above)
Comment (jhogue) [inline component/_footer.scss]: Every A should have a touch hover inside of it.

File: base/_typography.scss (hunk 7 above)
Comment (jhogue) [inline base/_typography.scss:90]: Limited, yes.

File: base/_typography.scss (hunk 8 above)
Comment (jhogue) [inline base/_typography.scss:279]: I'm not sure that bold always translates to the same numerical weight for every browser. Maybe this should be a rule that we agree on… always use the numeric weight esp. when using Google fonts.

Comment (jhogue) [inline base/_typography.scss:279]: https://css-tricks.com/almanac/properties/f/font-weight/

File: base/_typography.scss (hunk 11 above)
Comment (jhogue) [inline base/_typography.scss:368]: Style guide. We could use it, but best to apply any specific container classes to the vertical rhythm, which is now inside `_content.scss` as `.content--main { etc }`

File: base/_variables.scss (hunk 12 above)
Comment (jhogue) [inline base/_variables.scss:1]: I think that yes, there are some that deserve to be used more globally. Something like font-family could be used in many other places other than just the `typography.scss` file. We have all colors broken out into its own file, so now I think we need a place to put the rest of the global variables. You had an empty file in the scaffold before, I was trying to put a few in there that made sense.

File: component/_buttons.scss (hunk 13 above)
Comment (jhogue) [inline component/_buttons.scss:38]: I'm open to that.

**Comment by jhogue** [comment]
Grey Matter is CMS independent. With BEM in place, I'd like to see us simply change the container classes as needed once stuff goes from Grey Matter into a CMS. But even more, I'd like to see us impose our BEM conventions to the CMS of choice as much as possible.

**Comment by johncionci** [comment]
@jhogue Going to work on a mixin for use with 'graymatter'. I know that we want to employ BEM but I think that its hard if we are not maintaining the theme as we wont have too much control over rendered markup without programming. My thoughts are that BEM is for stuff that we do "custom" - or do you feel that its worth it for us to go down the path of creating filters (wp) and hooks (drupal) that actually modify the output code? if not, 

Im think something like this, maybe in some sort of "helpers" file:

```
@function navigation( $list-container, $list-tem, $list-link) {
ul.#{$list-container} {
  $list-container values...
li.#{$list-item} {
  $list-item values...
a.#{$list-link} {
  $list-link values...
}
}
}
}
```

It help be more agnostic... thoughts?

**Comment by jhogue** [comment]
I feel like that would be too granular and make the reading of the code
(maintainability) more difficult. Too much abstraction can be a hinderance.
If filters and hooks are the only way that we can fully control markup,
then I am not sure if it is worth it.

The goal is fewer long, specific selectors, like `.panels-page
.panels--quote blockquote p`. Bad example perhaps. But maybe that means
that we avoid nesting classes in SASS as much as possible if we can not
rely on BEM.

**Comment by johncionci** [comment]
Unfortunately I think thats the only way that we would get the most out of WP and Drupal codebase. We can set the nesting limits for BEM & regular SASS in the linter if you want. But I do think that BEM should be used for anything custom. So what do you think the solution is for the default classes issue?

**Comment by jhogue** [comment]
I don't really know. I don't want to put too much into the Scaffold that
people need to rip out, at the same time, I'd like it to be able to do
SOMETHING when we first pull it down.

On Tue, Mar 29, 2016 at 12:53 PM, John Cionci notifications@github.com
wrote:

> Unfortunately I think thats the only way that we would get the most out of
> WP and Drupal codebase. We can set the nesting limits for BEM & regular
> SASS in the linter if you want. But I do think that BEM should be used for
> anything custom. So what do you think the solution is for the default
> classes issue?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/oomphinc/scss-scaffold/pull/7#issuecomment-202995821

## 

J. Hogue
76 Locust Street, Apt 2
Providence, RI 02906

---

Highchairdesign.com- Graphic Design by J.
theGrantAt250.com- Collaborative Office Space
ArtInRuins.com- Mill advocacy in Providence

---

**Comment by johncionci** [comment]
@jhogue Can we pull out the "default classes" stuff for now so we can push this through for now. BTW Im hot on the trail of some code that will hopefully BEM-ify our navigations.

**General question**: should that stuff actually live in GrayMatter?

**Comment by jhogue** [comment]
Yes.

**Comment by jhogue** [comment]
@johncionci I stripped most of these files to the bone

**Comment by johncionci** [comment]
@jhogue status update?

**Comment by jhogue** [comment]
All your requested changes are in there, I believe. I ripped out a bunch of
starter classes that are unnecessary. Haven't put those patterns into my
Greymatter but I will.

**Comment by johncionci** [comment]
Cool man, lets resolve conflicts and roll it in!

**Pull request merged**